File: BE-MyCarMaster/src/main/java/softeer/be_my_car_master/domain/option/Option.java
package softeer.be_my_car_master.domain.option;

import java.util.List;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import softeer.be_my_car_master.global.annotation.Domain;

@Domain
@Getter
@Builder
@AllArgsConstructor
public class Option {

	private Long id;
	private String name;
	private Category category;
	private String summary;
	private String description;
	private String imgUrl;
	private Integer price;
	private Integer ratio;
	private Boolean isSuper;
	private List<Option> subOptions;
	private Tag tag;

	public String getTagName() {
		return tag == null ? null : tag.getName();
	}

	public Boolean hasSubOption() {
		return isSuper;
	}

	public String getCategoryValue() {
		return category.getValue();
	}
}

File: iOS-MyCarMaster/MyCarMaster/MyCarMaster/Model/Entity/Option.swift
//
//  Option.swift
//  MyCarMaster
//
//  Created by SEUNGMIN OH on 2023/08/16.
//

import Foundation

struct Option: Hashable, Priceable {
    var model: String
    var id: Int
    var category: String
    var name: String
    var price: Int
    var ratio: Int
    var imgURL: URL?
    var summary: String?
    var description: String?
    var tag: String?
    var subOptions: [SubOption]
}

struct SubOption: Hashable {
    let name: String
    let imgURL: URL?
    let description: String?

    enum CodingKeys: String, CodingKey {
        case name = "name"
        case imgURL = "imgUrl"
        case description = "description"
    }
}
